[Service Representation] Document precedence order of peer attributes (#26520)
DataDog/documentation: content/en/tracing/services/inferred_services.md

**Note**: Peer attribute values that match IP address formats (for example, 127.0.0.1) are modified and redacted with `blocked-ip-address` to prevent unnecessary noise and tagging metrics with high-cardinality dimensions. As a result, you may encounter some `blocked-ip-address` services appearing as downstream dependencies of your instrumented services.

#### Precedence of peer tags

To assign the name to inferred entities, Datadog uses a specific order of precedence between peer tags, when entities are defined by a combination of multiple tags. 

Entity type | Order of precedence
-----------|----------------
Database | `peer.db.name` > `peer.aws.s3.bucket` (For AWS S3) / `peer.aws.dynamodb.table` (For AWS DynamoDB) / `peer.cassandra.contact.points` (For Cassandra) / `peer.couchbase.seed.nodes` (For Couchbase) > `peer.hostname` > `peer.db.system`

> Queue | `peer.messaging.destination` > `peer.kafka.bootstrap.servers` (for kafka) / `peer.aws.sqs.queue` (for AWS SQS) / `peer.aws.kinesis.stream` (For AWS Kinesis) > `peer.messaging.system`

Reviewer: ```suggestion
Queue | `peer.messaging.destination` > `peer.kafka.bootstrap.servers` (for Kafka) / `peer.aws.sqs.queue` (for AWS SQS) / `peer.aws.kinesis.stream` (For AWS Kinesis) > `peer.messaging.system`
```

Inferred service | `peer.service` > `peer.rpc.service` > `peer.hostname`

> If the highest priority tag is not captured as part of the instrumentation (e.g. `peer.db.name`), Datadog uses the second highest priority tag (e.g. `peer.hostname`), and so on.

Reviewer: ```suggestion
If the highest priority tag, such as `peer.db.name`, is not captured as part of the instrumentation, Datadog uses the second highest priority tag, like `peer.hostname`, and continue in that order.
```

> **Note**: Datadog nevers sets the `peer.service` for inferred databases and queues. `peer.service` is the highest priority peer attribute. If set, it take precedence over the rest.

Reviewer: ```suggestion
**Note**: Datadog never sets the `peer.service` for inferred databases and queues. `peer.service` is the highest priority peer attribute. If set, it take precedence over all other attributes.
```
